File: src/rules/error-disclosure.js
const express = require('express');
const app = express();

app.get('/a', (req, res) => {
  try {
    doWork();
  } catch (err) {
    // ruleid: patronus-error-stack-to-client-js
    res.status(500).send(err.stack);
  }
});

app.get('/b', (req, res) => {
  try {
    doWork();
  } catch (err) {
    // ruleid: patronus-error-stack-to-client-js
    res.json({ error: err.stack });
  }
});

app.get('/c', (req, res) => {
  try {
    doWork();
  } catch (err) {
    console.error(err);
    // ok: patronus-error-stack-to-client-js
    res.status(500).send('Internal Server Error');
  }
});

File: src/rules/error-disclosure.py
import traceback
import logging

logger = logging.getLogger(__name__)


def handler_bad():
    try:
        do_work()
    except Exception:
        # ruleid: clipeus-traceback-to-client-python
        return traceback.format_exc()


def handler_bad_tuple():
    try:
        do_work()
    except Exception:
        # ruleid: clipeus-traceback-to-client-python
        return traceback.format_exc(), 500


def handler_good():
    try:
        do_work()
    except Exception:
        logger.exception("request failed")
        # ok: clipeus-traceback-to-client-python
        return "Internal Server Error", 500
